pallets/xyk/src/lib.rs

```diff
@@ -955,13 +940,17 @@ impl<T: Trait> XykFunctionsTrait<T::AccountId> for Module<T> {
         ensure!(first_asset_id != second_asset_id, Error::<T>::SameAsset,);
 
         // Liquidity token amount calculation
-        let initial_liquidity = first_asset_amount
-            .checked_add(second_asset_amount)
-            .ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
+        let mut initial_liquidity = first_asset_amount / 2 + second_asset_amount / 2;
+        if initial_liquidity == 0 {
+            initial_liquidity = 1
```

Comment:
why not just +1?

Reply:
don't want to have +1, just a min of 1.
other reason is potential overflow max balance / 2 + max balance / 2 +1 = max balance +1